### PR TITLE
fix: add nil guard in auditext manager Create function

### PR DESCRIPTION
**Pull request by ipsitapp8**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

Added a nil check at the top of the `Create` function in `src/pkg/auditext/manager.go`.
If `audit` is nil, the function now returns early with `0, nil` instead of panicking.
Also added a unit test to cover the nil case.

### CODE REVIEW

**Comment by ipsitapp8** [comment]
Could a maintainer please add a release-note/bug label?

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/pkg/auditext/manager.go [hunk 1]
```diff
@@ -77,6 +77,9 @@ func (m *manager) Get(ctx context.Context, id int64) (*model.AuditLogExt, error)
 
 // Create ...
 func (m *manager) Create(ctx context.Context, audit *model.AuditLogExt) (int64, error) {
+	if audit == nil {
```
Comment (chlins) [inline src/pkg/auditext/manager.go:80]: In what scenarios would this audit be nil?

Comment (ipsitapp8) [reply src/pkg/auditext/manager.go:80]: I added this as a defensive check in case `audit` ends up being nil from upstream (like unexpected caller behavior or future changes), just to avoid any panic.
I can remove it if you'd prefer enforcing non-nil at the caller instead.

**Review by chlins — APPROVED** [review]
lgtm
